apps/zotonic_core/src/models/m_dispatch.erl

```diff
@@ -0,0 +1,109 @@
+%% @author Marc Worrell <marc@worrell.nl>
+%% @copyright 2025 Marc Worrell
+%% @doc Model for dispatching requests.
+%% @end
+
+%% Copyright 2025 Marc Worrell
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(m_dispatch).
+-author("Marc Worrell <marc@worrell.nl").
+
+-behaviour(zotonic_model).
+
+-export([
+    m_get/3,
+
+    dispatch_url/2,
+    dispatch_path/2
+]).
+
+m_get([ <<"url">> ], #{ payload := #{ <<"url">> := Url } }, Context) ->
+    dispatch_url(Url, Context);
+m_get([ <<"url">> ], _Msg, Context) ->
+    dispatch_url(z_context:get_q(<<"url">>, Context), Context);
+m_get([ <<"url">>, Url ], _Msg, Context) ->
+    dispatch_url(Url, Context);
```

Comment:
I'm confused about this one, how could we pass a URL to the model as a single path section? 🤔 

I suppose we could do something like for `dispatch_path`:
```suggestion
m_get([ <<"url">> | Url ], _Msg, Context) ->
    io:format("debug url: ~p~n", [Url]),
    dispatch_url(Url, Context);
```
and rebuild the URL in `dispatch_url`, but `z_sites_dispatcher:dispatch_url` requires the URL to be complete, including scheme and host, so I think it would be messy to use, e.g.
```
    curl -k 'https://example.test:8443/api/model/dispatch/get/url/https://example.test:8443/nl/page/1234/hello'
```

Unless I'm missing something, I'd suggest to remove this:
```suggestion
```

Reply:
You can pass it at once if you url encode it or use it as an argument in a template.  In that way it is a shorthand way:

```django
m.dispatch.url:%{ url: 'http://...' }
```

versus

```django
m.dispatch.url['http://...']
```